File: AjaxVectorObjects/ClientScripts/RedoUndo/LayerMovedCommand.js
// Copyright (c) 2018 Aurigma Inc. All rights reserved.
// Licensed under the MIT License. See LICENSE file in the project root for full license information.
//
Type.registerNamespace("Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo");

Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.LayerMovedCommand = function (layer, oldLayerIndex, newLayerIndex) {
    /// <summary>This class represents the layer moved command.</summary>
    /// <constructor>
    /// 	<exclude />
    /// </constructor>
    this.OldLayerIndex = new Number(oldLayerIndex);
    if (isNaN(this.OldLayerIndex))
        this.OldLayerIndex = -1;
    this.NewLayerIndex = new Number(newLayerIndex);
    if (isNaN(this.NewLayerIndex))
        this.NewLayerIndex = -1;
}

Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.LayerMovedCommand.prototype = {
    execute: function (canvas) {
        /// <summary>Executes this command.</summary>
        /// <param name="canvas" type="Aurigma.GraphicsMill.AjaxControls.VectorObjects.Canvas">The canvas to execute the command on.</param>
        canvas.get_layers().move(this.OldLayerIndex, this.NewLayerIndex);

        Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.LayerMovedCommand.callBaseMethod(this, 'execute');
    },

    unExecute: function (canvas) {
        /// <summary>Rolls this command back.</summary>
        /// <param name="canvas" type="Aurigma.GraphicsMill.AjaxControls.VectorObjects.Canvas">The canvas to roll the command back.</param>
        canvas.get_layers().move(this.NewLayerIndex, this.OldLayerIndex);

        Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.LayerMovedCommand.callBaseMethod(this, 'unExecute');
    }
}

Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.LayerMovedCommand.registerClass("Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.LayerMovedCommand",
	Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.Command);

File: AjaxVectorObjects/Common/RedoUndo/LayerMovedCommand.cs
// Copyright (c) 2018 Aurigma Inc. All rights reserved.
// Licensed under the MIT License. See LICENSE file in the project root for full license information.
//

namespace Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo
{
    internal class LayerMovedCommand : Command
    {
        private int _oldLayerIndex = -1;
        private int _newLayerIndex = -1;

        public LayerMovedCommand(Layer layer, int oldIndex, int newIndex)
        {
            _oldLayerIndex = oldIndex;
            _newLayerIndex = newIndex;
        }

        public int OldLayerIndex
        {
            get { return _oldLayerIndex; }
            set { _oldLayerIndex = value; }
        }

        public int NewLayerIndex
        {
            get { return _newLayerIndex; }
            set { _newLayerIndex = value; }
        }

        public override void Execute(ICanvas canvas)
        {
            canvas.Layers.Move(this._oldLayerIndex, this._newLayerIndex);
        }

        public override void UnExecute(ICanvas canvas)
        {
            canvas.Layers.Move(this._newLayerIndex, this._oldLayerIndex);
        }
    }
}